infra/base-images/msan-builder/msan_build.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+from __future__ import print_function
+import argparse
+import os
+import shutil
+import subprocess
+import tempfile
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__)) 
+
+class MSanBuildException(Exception):
+  """Base exception."""
+
+
+def SetUpEnvironment(work_dir):
+  """Set up build environment."""
+  env = {}
+  env['REAL_CLANG_PATH'] = subprocess.check_output(['which', 'clang']).strip()
+  print('Real clang at', env['REAL_CLANG_PATH'])
+  compiler_wrapper_path = os.path.join(SCRIPT_DIR, 'compiler_wrapper.py')
+
+  # Symlink binaries into TMP/bin
+  bin_dir = os.path.join(work_dir, 'bin')
+  os.mkdir(bin_dir)
+
+  os.symlink(compiler_wrapper_path,
+             os.path.join(bin_dir, 'clang'))
+
+  os.symlink(compiler_wrapper_path,
+             os.path.join(bin_dir, 'clang++'))
+
+  env['CC'] = os.path.join(bin_dir, 'clang')
+  env['CXX'] = os.path.join(bin_dir, 'clang++')
+
+  # Not all build rules respect $CC/$CXX, so make additional symlinks.
```

Comment:
would it also make sense to try setting other "popular" env variables like `CPP`, `cc`, `cpp`, `cxx`?

Reply:
Hmm, let's stick with CC/CXX for now and see if we miss any.